### PR TITLE
Fix offset-based position transform not respecting previous transforms

### LINKED ISSUE #3821: MoveToOffset() does not respect previous position transforms on all cases
Realized it on #3817, probably unnoticed due to its lack of usage.

Since `MoveToOffset()` is an "offset transformation", the end value depends on the current one, the way it's currently done is by checking the last existing `Position` transform:
https://github.com/ppy/osu-framework/blob/a949b10f85ade1137b63e61cbb48bb32aa781834/osu.Framework/Graphics/TransformableExtensions.cs#L666

which would not work on all cases, such as using `MoveTo[X,Y]()` transforms before instead:
```csharp
            boxTest(box =>
            {
                box.MoveToY(0.25f, 250);
                box.Delay(500).MoveToOffset(new Vector2(0, 0.25f), 250);
            });

            checkAtTime(0, box => box.Y == 0);
            checkAtTime(250, box => box.Y == 0.25f);
            checkAtTime(500, box => box.Y == 0.25f);
            checkAtTime(750, box => box.Y == 0.5f);
```
or not adding the transforms in order:
```csharp
            boxTest(box =>
            {
                box.Delay(500).MoveToOffset(new Vector2(0, 0.25f), 250);
                box.MoveToOffset(new Vector2(0, 0.25f), 250);
            });

            checkAtTime(0, box => box.Y == 0);
            checkAtTime(250, box => box.Y == 0.25f);
            checkAtTime(500, box => box.Y == 0.25f);
            checkAtTime(750, box => box.Y == 0.5f);
```

(above cases can be testable my inserting them into `TestSceneTransformRewinding` each in a `Test` method).

---

Reporting this for tracking, didn't wanna miss this one as well.

### PR DESCRIPTION
**Pull request by frenzibyte**
Closes #3821
Supersedes #3902 

See https://github.com/ppy/osu-framework/pull/3902#issuecomment-699742964 for reason why superseded.

---

`ReadIntoStartValue()` is added before `Apply()` to fix cases where the position offset transform would be applied without its value being read, therefore leading to incorrect results. (as the commit says https://github.com/ppy/osu-framework/commit/1589926137f8e295bad6923bd939ea8af96d9c43)

### CODE REVIEW

File: osu.Framework/Graphics/Transforms/TargetGroupingTransformTracker.cs [hunk 1]
```diff
@@ -303,6 +303,7 @@ public virtual void FinishTransforms(string targetMember = null)
 
             foreach (Transform t in toFlush)
             {
+                t.ReadIntoStartValue();
```
Comment (smoogipoo) [inline osu.Framework/Graphics/Transforms/TargetGroupingTransformTracker.cs:306]: Surround with `if (!t.HasStartValue) { ...; t.HasStartValue = true; }`

**Review by smoogipoo — CHANGES_REQUESTED** [review]

File: osu.Framework/Graphics/Transforms/TargetGroupingTransformTracker.cs (hunk 1 above)
Comment (frenzibyte) [reply osu.Framework/Graphics/Transforms/TargetGroupingTransformTracker.cs:306]: oh right, I thought the transform would be removed, but that would not be the case for rewinding

**Review by smoogipoo — APPROVED** [review]

**Comment by smoogipoo** [comment]
@peppy This seems sane to me but I want your review too since you're more knowledgeable about the inner workings of transforms w.r.t reading of values/applying.

**Comment by peppy** [comment]
I think it seems sane. @smoogipoo have you cross-tested with osu! (especially gameplay/storyboard rewinding/ffwding)?

**Review by peppy — APPROVED** [review]

**Comment by smoogipoo** [comment]
Just did, seems fine.

**Pull request merged**